report/marshal.go

```diff
@@ -56,13 +55,7 @@ func (rep *Report) ReadBinary(r io.Reader, gzipped bool, codecHandle codec.Handl
 	if log.GetLevel() == log.DebugLevel {
 		r = byteCounter{next: r, count: &uncompressedSize}
 	}
-	var decoder func(interface{}) error
-	if codecHandle != nil {
-		decoder = codec.NewDecoder(r, codecHandle).Decode
-	} else {
-		decoder = gob.NewDecoder(r).Decode
```

Comment:
Awesome step forward for https://github.com/weaveworks/scope/issues/971. It's a larger task but it would be good to remove all the GobEncode/GobDecode methods  (probably in a separate PR).